src/plugins/embeddable/public/lib/embeddables/i_embeddable.ts

```diff
@@ -22,6 +22,7 @@ import { Adapters } from '../types';
 import { IContainer } from '../containers/i_container';
 import { ViewMode } from '../types';
 import { TriggerContextMapping } from '../../../../ui_actions/public';
+import { TimeRange, Query, Filter } from '../../../../data/common';
```

Comment:
Maybe let's make it explicit as `type` import ? 

```suggestion
import type { TimeRange, Query, Filter } from '../../../../data/common';
```

Reply:
Done.